tests/unit/states/test_powerdns.py

```diff
@@ -0,0 +1,106 @@
+        )
+
+    def test_manage_zone_path_empty_name(self):
+        ret = {
+            "name": "",
+            "changes": {},
+            "result": False,
+            "comment": "No name of zone provided",
+        }
+        self.assertEqual(powerdns.manage_zone("", "key", "server"), ret)
+
+    def test_manage_zone_was_updated(self):
+        result_changes = {"changes": "Zone was updated"}
+        mock_powerdns_modules = {
+            "powerdns.manage_zone": MagicMock(return_value=result_changes)
+        }
+        with patch.dict(powerdns.__salt__, mock_powerdns_modules):
+            ret = {
+                "name": "zone",
+                "changes": "Zone was updated",
+                "result": True,
+                "comment": "",
+            }
+            self.assertEqual(powerdns.manage_zone("zone", "key", "server"), ret)
+
+    def test_manage_zone_failed(self):
+        mock_powerdns_modules = {
+            "powerdns.manage_zone": MagicMock(side_effect=[CommandExecutionError])
+        }
+        with patch.dict(powerdns.__salt__, mock_powerdns_modules):
+            self.assertRaises(
+                CommandExecutionError,
+                powerdns.manage_zone,
+                "test_domain",
+                "key",
+                "server",
+            )
+
+    def test_delete_zone_path_empty_key(self):
+        self.assertRaises(
+            SaltInvocationError, powerdns.delete_zone, "test_domain", "", "server"
+        )
+
+    def test_delete_zone_path_empty_server(self):
+        self.assertRaises(
+            SaltInvocationError, powerdns.delete_zone, "test_domain", "key", ""
+        )
+
+    def test_delete_zone_path_empty_name(self):
+        ret = {
+            "name": "",
+            "changes": {},
+            "result": False,
+            "comment": "No name of zone provided",
+        }
+        self.assertEqual(powerdns.delete_zone("", "key", "server"), ret)
+
+    def test_delete_zone_deleted(self):
+        result_changes = {"changes": "Zone was deleted"}
+        mock_powerdns_modules = {
+            "powerdns.delete_zone": MagicMock(return_value=result_changes)
+        }
+        with patch.dict(powerdns.__salt__, mock_powerdns_modules):
+            ret = {
+                "name": "zone",
+                "changes": "Zone was deleted",
+                "result": True,
+                "comment": "",
+            }
+            self.assertEqual(powerdns.delete_zone("zone", "key", "server"), ret)
+
+    def test_delete_zone_failed(self):
+        mock_powerdns_modules = {
+            "powerdns.delete_zone": MagicMock(side_effect=[CommandExecutionError])
+        }
+        with patch.dict(powerdns.__salt__, mock_powerdns_modules):
+            self.assertRaises(
+                CommandExecutionError,
+                powerdns.delete_zone,
+                "test_domain",
+                "key",
+                "server",
+            )
```

Comment:
Since this is a new test module, would you mind rewriting it in `pytest` under `tests/pytests/unit/states`?

Reply:
Rewrote.

Reply:
Thank You!